statescript/executor.go

```diff
@@ -206,13 +240,31 @@ func (l Launcher) ExecuteAll(state, action string, ignoreError bool) error {
 			}
 		}
 
-		if ret := execute(filepath.Join(dir, s.Name()), timeout); ret != 0 {
-			// In case of error scripts all should be executed.
-			if ignoreError {
-				log.Errorf("statescript: ignoring error executing '%s': %d", s.Name(), ret)
-			} else {
-				return errors.Errorf("statescript: error executing '%s': %d",
-					s.Name(), ret)
+		iet := time.Now()
+
+	retrylater:
+		for {
+			switch ret := execute(filepath.Join(dir, s.Name()), timeout); ret {
+			case 0:
+				// success. Break the loop and continue
+				break retrylater
+			case exitRetryLater:
+				if time.Since(iet) <= l.getRetryTimeout() {
```

Comment:
Not sure why do we need this check here and the timeout. Can't we simply read return code and if script is requesting retrying later we should wait the retry interval?